templates/repo/commits_list.tmpl

```diff
@@ -72,6 +72,9 @@
 						{{if IsMultilineCommitMessage .Message}}
 						<pre class="commit-body tw-hidden">{{RenderCommitBody $.Context .Message ($.Repository.ComposeMetas ctx)}}</pre>
 						{{end}}
+						{{range (index $.CommitsTagsMap .ID.String)}}
+							<span class="ui label">{{.TagName}}</span>
```

Comment:
I recommend three things about this:
1. extract it into a subtemplate (although I know that one is controversial, some maintainers prefer subtemplates while others prefer a single template)
2. Make it a link that links either to the release or tag
3. Highlight tags with a release in a different color, i.e. releases are colored in the primary color while tags without release are only grey or something like that

Reply:
And perhaps 4: Give it only minimal width to display perhaps three chars or so, and display the rest of the name in a tooltip.

Reply:
1,2,3 done. I don't think 4 is necessary because commonly the tag name is not too long.

Reply:
The problem is not the `commonly` but exactly the `uncommon`.
Especially on mobile